app/grandchallenge/notifications/migrations/0008_migrate_forum_follows_and_notifications.py

```diff
@@ -0,0 +1,111 @@
+from django.db import migrations
+
+from grandchallenge.notifications.models import NotificationTypeChoices
+
+
+def migrate_forum_and_topic_follows_and_notifications(apps, schema_editor):
+    Notification = apps.get_model(  # noqa: N806
+        "notifications", "Notification"
+    )
+    Follow = apps.get_model("actstream", "Follow")  # noqa: N806
+    Forum = apps.get_model("discussion_forums", "Forum")  # noqa: N806
+    ForumTopic = apps.get_model(  # noqa: N806
+        "discussion_forums", "ForumTopic"
+    )
+    MachinaForum = apps.get_model("forum", "Forum")  # noqa: N806
+    MachinaTopic = apps.get_model("forum_conversation", "Topic")  # noqa: N806
+    ContentType = apps.get_model("contenttypes", "ContentType")  # noqa: N806
+
+    new_topic_ct = ContentType.objects.get_for_model(ForumTopic)
+    new_forum_ct = ContentType.objects.get_for_model(Forum)
+
+    def get_matching_forum(*, old_forum_id):
+        old_forum = MachinaForum.objects.get(pk=old_forum_id)
+        return old_forum.challenge.discussion_forum
+
+    def get_matching_topic(*, old_topic_id):
+        old_topic = MachinaTopic.objects.get(pk=old_topic_id)
+        new_forum = get_matching_forum(old_forum_id=old_topic.forum.pk)
+        return ForumTopic.objects.get(
+            forum=new_forum,
+            creator=old_topic.poster,
+            subject=old_topic.subject,
+        )
+
+    notifications_to_update = []
+    for notification in Notification.objects.filter(
```

Comment:
This is going to go OOM, we have 600k post notifications and 7k reply notifications, modifying that many objects will take several hours even with batching. I don't know what the time limit is on the ECS task. See the utilization app for some strategy. 1000 objects fetched even when using `only` still took up 25% of the memory.

Maybe we just have an amnesty and delete the existing notifications?

Reply:
Right, I thought of batching when doing the bulk update later, but forgot that this is of course reads them all into memory as well.... 

Deleting seems ok to me in this case. I doubt people would miss old notifications. Arguably, the value of a notification disappears once it's been read...

Reply:
However, even if we delete objects, we still need to batch process them to avoid OOM.

Reply:
You can do `queryset.delete()`, does everything db side.